src/main/scala/EnsimePlugin.scala

```diff
@@ -561,6 +599,29 @@ case class EnsimeModule(
 
 }
 
+case class EnsimeProject(
+  name: String,
+  dependsOnNames: Set[String],
```

Comment:
@fommil do you think something like `dependsOn: Map[String, String]` with 
`Map(foo -> compile, foo -> test, bar -> "compile")` is a good idea